src/App.tsx

```diff
@@ -1,29 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+export type Page = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+};
+
 export const App: React.FC = () => {
+  const perPageSelectors = [3, 5, 10, 20];
+  const [perPage, setPerPage] = useState(perPageSelectors[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const listStartsWith = perPage * (currentPage - 1);
+  const listEndsWith = Math.min(listStartsWith + perPage - 1, items.length - 1);
+  const description = `Page ${currentPage} (items ${listStartsWith + 1} - ${listEndsWith + 1} of ${items.length})`;
+
+  const itemsList = [...items].slice(listStartsWith, listEndsWith + 1);
```

Comment:
```suggestion
  const itemsList = items.slice(listStartsWith, listEndsWith + 1);
```
slice doesn't mutate the array